router/default.go

```diff
@@ -436,6 +446,12 @@ func (r *router) advertiseEvents() error {
 				delete(adverts, key)
 			}
 
+			// If we're not advertising any events then throw them away.
+			// We do this here because someone may change the strategy at any time.
+			if r.options.Advertise == AdvertiseNone {
```

Comment:
Shouldnt we call this before we start iterating over the adverts?

Reply:
Done